data/src/test/java/org/apache/iceberg/data/TestPartitionStatsHandler.java

```diff
@@ -0,0 +1,569 @@
+            required(111, "uuid", Types.UUIDType.get()),
+            required(112, "fixed", Types.FixedType.ofLength(7)),
+            optional(113, "bytes", Types.BinaryType.get()),
+            required(114, "dec_9_0", Types.DecimalType.of(9, 0)),
+            required(115, "dec_11_2", Types.DecimalType.of(11, 2)),
+            required(116, "dec_38_10", Types.DecimalType.of(38, 10)), // maximum precision
+            required(117, "time", Types.TimeType.get()));
+
+    PartitionSpec spec =
+        PartitionSpec.builderFor(schema)
+            .identity("b")
+            .identity("i")
+            .identity("l")
+            .identity("f")
+            .identity("d")
+            .identity("date")
+            .identity("ts")
+            .identity("s")
+            .identity("uuid")
+            .identity("fixed")
+            .identity("bytes")
+            .identity("dec_9_0")
+            .identity("dec_11_2")
+            .identity("dec_38_10")
+            .identity("time")
+            .build();
+
+    Table testTable =
+        TestTables.create(
+            tempDir("test_all_type"), "test_all_type", schema, spec, SortOrder.unsorted(), 2);
+
+    Types.StructType partitionSchema = Partitioning.partitionType(testTable);
+    Schema dataSchema = PartitionStatsHandler.schema(partitionSchema);
+
+    Record partitionData =
+        GenericRecord.create(dataSchema.findField(Column.PARTITION.name()).type().asStructType());
+    partitionData.set(0, true);
+    partitionData.set(1, 42);
+    partitionData.set(2, 42L);
+    partitionData.set(3, 3.14f);
+    partitionData.set(4, 3.141592653589793);
+    partitionData.set(5, Literal.of("2022-01-01").to(Types.DateType.get()).value());
+    partitionData.set(
+        6, Literal.of("2017-12-01T10:12:55.038194").to(Types.TimestampType.withoutZone()).value());
+    partitionData.set(7, "string");
+    partitionData.set(8, UUIDUtil.convertToByteBuffer(UUID.randomUUID()));
+    partitionData.set(9, new byte[] {0, 1, 2, 3, 4, 5, 6});
+    partitionData.set(10, new byte[] {1, 2, 3});
+    partitionData.set(11, Literal.of("123456789").to(Types.DecimalType.of(9, 0)).value());
+    partitionData.set(12, Literal.of("1234567.89").to(Types.DecimalType.of(11, 2)).value());
+    partitionData.set(
+        13, Literal.of("12345678901234567890.1234567890").to(Types.DecimalType.of(38, 10)).value());
+    partitionData.set(14, Literal.of("10:10:10").to(Types.TimeType.get()).value());
+
+    PartitionStats partitionStats = new PartitionStats(partitionData, RANDOM.nextInt(10));
+    partitionStats.set(Column.DATA_RECORD_COUNT.id(), RANDOM.nextLong());
+    partitionStats.set(Column.DATA_FILE_COUNT.id(), RANDOM.nextInt());
+    partitionStats.set(Column.TOTAL_DATA_FILE_SIZE_IN_BYTES.id(), 1024L * RANDOM.nextInt(20));
+
+    Iterator<PartitionStatsRecord> convertedRecords =
+        PartitionStatsHandler.statsToRecords(Collections.singletonList(partitionStats), dataSchema);
```

Comment:
Can we add a few more records here?

Reply:
There are already other tests below (for example `testPartitionStats`) that checks multiple rows/records. 

This testcase is to check one row of "all possible data types" in partition values. I don't see the benefit of having more rows here.